R/scale-date.r

```diff
@@ -197,20 +256,35 @@ scale_datetime <- function(aesthetics, trans,
     date = "date",
     time = "datetime"
   )
-  scale_class <- switch(trans,
-    date = ScaleContinuousDate,
-    time = ScaleContinuousDatetime
-  )
+
+  # x/y position aesthetics should use ScaleContinuousDate or
+  # ScaleContinuousDatetime; others use ScaleContinuous
+  if (all(aesthetics %in% c("x", "xmin", "xmax", "xend", "y", "ymin", "ymax", "yend"))) {
```

Comment:
It's so the data will get rescaled correctly, I think. `ScaleContinuous` uses `scales::rescale()` to map the data values to the right place on the scale. For position scales this isn't needed because the coordinate system handles it.